lib/tbd/geo.rb

```diff
@@ -756,33 +765,44 @@ def kiva(model = nil, walls = {}, floors = {}, edges = {})
     mth = "TBD::#{__callee__}"
     cl1 = OpenStudio::Model::Model
     cl2 = Hash
-    kva = true
     a   = false
     return mismatch("model" ,  model, cl1, mth, DBG, a) unless model.is_a?(cl1)
     return mismatch("walls" ,  walls, cl2, mth, DBG, a) unless walls.is_a?(cl2)
     return mismatch("floors", floors, cl2, mth, DBG, a) unless floors.is_a?(cl2)
     return mismatch("edges" ,  edges, cl2, mth, DBG, a) unless edges.is_a?(cl2)
 
+    # Check for existing KIVA objects.
+    kva = false
+    kva = true unless model.getSurfacePropertyExposedFoundationPerimeters.empty?
+    kva = true unless model.getFoundationKivas.empty?
+
+    if kva
+      log(ERR, "Exiting - KIVA objects in model (#{mth})")
+      return a
+    else
+      kva = true
+    end
```

Comment:
Both (higher level) _measure.rb_ and _psi.rb_ hold optional calls to the **resetKIVA** method, prior to calling TBD's (lower level) **kiva** method. The latter is not as forgiving and exits with an error if existing KIVA objects are found in the model.